version: '3.4'

services:
  hyre-bootstrapper:
    image: ${DOCKER_REGISTRY-}hyrebootstrapper:latest
    container_name: hyre-api
    build:
      context: .
      dockerfile: src/Bootstrapper/Hyre.Bootstrapper/Dockerfile
    depends_on:
      hyre-database:
        condition: service_healthy
    networks:
      - hyre

  hyre-database:
    image: postgres
    container_name: hyre-db
    hostname: hyre-db
    shm_size: '4gb'
    restart: unless-stopped
    environment:
      POSTGRES_HOST_AUTH_METHOD: trust
      POSTGRES_PASSWORD: postgres
      POSTGRES_USER: postgres
      POSTGRES_DB: hyre-db
    ports:
      - "5432:5432"
    volumes:
      - ./containers/pg-data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - hyre
    depends_on: 
      hyre-message-broker:
        condition: service_healthy
  
  hyre-message-broker:
    image: rabbitmq:management-alpine
    container_name: hyre-message-broker
    hostname: hyre-message-broker
    restart: unless-stopped
    environment:
      RABBITMQ_DEFAULT_USER: hyre
      RABBITMQ_DEFAULT_PASS: hyre123
    volumes:
      - ./containers/rabbitmq/data:/var/lib/rabbitmq
      - ./containers/rabbitmq/logs:/var/log/rabbitmq
    ports:
      - "5672:5672"
      - "15672:15672"
    networks:
      - hyre
    healthcheck:
      test: ["CMD", "rabbitmqctl", "status"]
      interval: 10s
      timeout: 5s
      retries: 5

volumes:
  pg-data:
    driver: local

networks:
  hyre:
    external: true
    name: hyre-network